### PR TITLE
[WIP] [SWA-208]: Creates external_id column in submissions table

### PR DESCRIPTION
**Pull request by nickskalkin**
Creates external_id column in submissions table.

Jira card - https://artsyproduct.atlassian.net/jira/software/c/projects/SWA/boards/113?modal=detail&selectedIssue=SWA-208

### CODE REVIEW

File: db/migrate/20220125100834_add_external_id_to_submissions.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class AddExternalIdToSubmissions < ActiveRecord::Migration[6.1]
+  def change
+    enable_extension 'pgcrypto' unless extension_enabled?('pgcrypto') # need to enable pgcrypto to use `gen_random_uuid()` function
```
Comment (nickskalkin) [inline db/migrate/20220125100834_add_external_id_to_submissions.rb:3]: Had to enable `pgcrypto`, otherwise `gen_random_uuid()` wouldn't work

**Review by Gon4ar5 — APPROVED** [review]
LGTM!

**Review by kiryl-zubarau — APPROVED** [review]

File: db/migrate/20220125100834_add_external_id_to_submissions.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+    add_column :submissions,
+               :external_id,
```
Comment (iskounen) [inline db/migrate/20220125100834_add_external_id_to_submissions.rb:6]: **question:** What will `external_id` represent? My understanding is that:

- `id` will continue being the sequential, numerical ID
- `uuid` will be the non-sequential, generated ID

I assumed that we would expose the `uuid` field to clients as an "external" ID so I am wondering why we have a separate `external_id` column.

Comment (nickskalkin) [reply db/migrate/20220125100834_add_external_id_to_submissions.rb:6]: @iskounen `external_id` is basically a `uuid` column you are talking about. Clients will be working with submissions using this column. Do you propose to rename `external_id` to `uuid`? I actually like `uuid` more

Comment (iskounen) [reply db/migrate/20220125100834_add_external_id_to_submissions.rb:6]: Thanks for the explanation. I was reading the migration file incorrectly... 😅 

Regarding the name, my preference is to use `uuid` on the data model, and decide how we want to expose that to clients at the GraphQL layer.

Comment (nickskalkin) [reply db/migrate/20220125100834_add_external_id_to_submissions.rb:6]: I will move this PR to WIP and rename db column. @kiryl-zubarau @iskounen what is your opinion on how to name this field in API?

Comment (kiryl-zubarau) [reply db/migrate/20220125100834_add_external_id_to_submissions.rb:6]: I would prefer `external id`

**Comment by nickskalkin** [comment]
I am closing this PR in favour of https://github.com/artsy/convection/pull/1224

The second one is based on the current one. I had to make changes in the migration and I decided to make in the second PR,  just because it is easier. Let me know if you still prefer that I make changes in this PR and align history of the second one

cc @kiryl-zubarau @iskounen

**Pull request closed without merging**